src/main/java/net/apnic/rdap/rdap/RDAPError.java

```diff
@@ -0,0 +1,98 @@
+package net.apnic.rdap.rdap;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Class represent an error response in the RDAP protocol.
+ *
+ * @see https://tools.ietf.org/html/rfc7483
+ */
+public class RDAPError
+    extends RDAPObject
+{
+    private List<String> description = null;
```

Comment:
Thats fine but now everything that ever wants an RDAPError or subclass of RDAPObject now needs to know how to construct no only the child type but the parent type and load it with defaults.

It just seems to me that doing it this way requires any using code now to come must have knowledge of all the elements that go into constructing a valid RDAPObject for a response.

You could make an RDAPDefaults object that can be passed into the constructor and passed up to the parent object. This method is ok but its still forcing everything that wants to construct an RDAPObject subclass to know about the existence of defaults to construct a valid response.

In my opinion it seems we would just be trading one thing for another and still not realising a perfect design.